db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/SYSTABLESRowFactory.java

```diff
@@ -52,661 +52,664 @@
+
+        // RESOLVE - Deal with lock granularity
+        tabDesc = ddg.newTableDescriptor(tableName, schema, tableTypeEnum, lockGranularity.charAt(0),
+                row.getColumn(SYSTABLES_COLUMN_SEQUENCE).getInt(),
+                delimitedDVD != null ? delimitedDVD.getString() : null,
+                escapedDVD != null ? escapedDVD.getString() : null,
+                linesDVD != null ? linesDVD.getString() : null,
+                storedDVD != null ? storedDVD.getString() : null,
+                locationDVD != null ? locationDVD.getString() : null,
+                compressionDVD != null ? compressionDVD.getString() : null,
+                isPinnedDVD.getBoolean(),
+                purgeDeletedRowsDVD.getBoolean(),
+                minRetentionPeriodDVD.getLong()
+        );
+        tabDesc.setUUID(tableUUID);
+
+        if (versionDescriptor != null) {
+            tabDesc.setVersion(versionDescriptor.getString());
+        } else
+            tabDesc.setVersion(ORIGINAL_TABLE_VERSION);
+        return tabDesc;
+    }
+
+    /**
+     * Get the table name out of this SYSTABLES row
+     *
+     * @param row a SYSTABLES row
+     * @throws StandardException thrown on failure
+     * @return string, the table name
+     */
+    protected String getTableName(ExecRow row)
+            throws StandardException {
+        DataValueDescriptor col;
+
+        col = row.getColumn(SYSTABLES_TABLENAME);
+        return col.getString();
+    }
+
+
+    /**
+     * builds a list of columns suitable for creating this catalog.
+     *
+     * @return array of systemcolumn suitable for making this catalog.
+     */
+    public SystemColumn[] buildColumnList()
+            throws StandardException {
+        return new SystemColumn[]{
+                SystemColumnImpl.getUUIDColumn(TABLEID, false),
+                SystemColumnImpl.getIdentifierColumn(TABLENAME, false),
+                SystemColumnImpl.getIndicatorColumn(TABLETYPE),
+                SystemColumnImpl.getUUIDColumn(SCHEMAID, false),
+                SystemColumnImpl.getIndicatorColumn(LOCKGRANULARITY),
+                SystemColumnImpl.getIdentifierColumn(VERSION, true),
+                SystemColumnImpl.getColumn(COLSEQUENCE, Types.INTEGER, false),
+                SystemColumnImpl.getColumn(DELIMITED, Types.VARCHAR, true),
+                SystemColumnImpl.getColumn(ESCAPED, Types.VARCHAR, true),
+                SystemColumnImpl.getColumn(LINES, Types.VARCHAR, true),
+                SystemColumnImpl.getColumn(STORED, Types.VARCHAR, true),
+                SystemColumnImpl.getColumn(LOCATION, Types.VARCHAR, true),
+                SystemColumnImpl.getColumn(COMPRESSION, Types.VARCHAR, true),
+                SystemColumnImpl.getColumn(IS_PINNED, Types.BOOLEAN, false),
                 SystemColumnImpl.getColumn(PURGE_DELETED_ROWS, Types.BOOLEAN, false),
+                SystemColumnImpl.getColumn(MIN_RETENTION_PERIOD, Types.BIGINT, false),
```

Comment:
We may need to define MIN_RETENTION_PERIOD as nullable, since the existing rows upgraded from lower versions won't have this field, so it's value is null.

Reply:
done.